### PR TITLE
Add Infrastructure Helm Chart

### PR DESCRIPTION
**Pull request by shaharblanksela**
Add a new chart 'infra-chart' which includes shared resources for all microservices:
PVC for efs.

Bug: Argocd can't recognize nested charts inside child-chart in umbrella-chart

helm/charts/amplication
├── charts
│   ├── amplication-client
│   ├── amplication-git-pull-request-service
│   ├── amplication-git-pull-service
│   ├── amplication-git-push-webhook-service
│   ├── amplication-server
│   ├── amplication-storage-gateway
│   ├── base
│   ├── infra 
in infra:   │   ├── cp-helm-charts (kafka & zookeeper charts)    ❌
                │   ├── postgresql chart    ❌
                │   ├── pvc.yaml ✅

After searching online and discussing it with a colleague, I couldn't find a valid solution for this.
The previous approach: Kafka chart & Postgres are located in the umbrella chart and the variables to override are in values-common.yaml file. 

I would like to get another opinion on the subject.

### CODE REVIEW

**Review by arielweinberger — APPROVED** [review]
Looks great! I'd wait for a review from @mshidlov since I am lacking some context.

File: helm/charts/services/amplication-server/values.yaml [hunk 1]
```diff
@@ -38,11 +38,6 @@ minMemory: "2Gi"
...
 minCPU: 1
 
-pvc:
-  storageLimit: 5Gi
-  storageRequest: 1Gi
-  storageName: efs-sc
-  name: efs-claim
 
 # Postgresql Server
 postgresql:
```
Comment (mshidlov) [inline helm/charts/services/amplication-server/values.yaml:43]: This should be taken from the infra chart

**Review by mshidlov — CHANGES_REQUESTED** [review]

File: helm/charts/amplication/values-amit.yaml [hunk 2]
```diff
@@ -1,16 +1,8 @@
...
-postgres:
-  install: true
-  postgresqlPassword: postgres
-  postgresqlPostgresPassword: postgres
-  postgresqlDatabase: amplication
-  fullnameOverride: postgresql
+global:
+  pvc:
+    name: amit-efs-claim
+  pv:
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/amplication/values-amit.yaml:4]: Yes, they are. PV is a resource outside the namespace which connects the PVC request from volume data to be stored in efs.  volume is the name of the folder located in the pod.

File: helm/charts/amplication/values-common.yaml [hunk 3]
```diff
@@ -63,17 +74,15 @@ amplication-client:
...
     install: false
   ingress:
     enabled: true
     hostname: server.staging.amplication-dev.com
-  pvc:
-    name: common-efs-claim
   config:
     env:
-      GENERATE_PULL_REQUEST_TOPIC: git.internal.pull-request.request.0
+      GENERATE_PULL_REQUEST_TOPIC: git.internal.pull-request.request.1
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/amplication/values-common.yaml:85]: Changed. I pulled the folder of the helm chart from my previous branch and this is a value from it.  not relevant changed it to '0'.

File: helm/charts/amplication/values-common.yaml [hunk 4]
```diff
@@ -167,22 +174,22 @@ amplication-git-pull-service:
...
-  pvc:
-    name: common-efs-claim
+    tag: "release-0-13-1"
   config:
     env:
       KAFKA_BROKERS: '["amplication-cp-kafka-headless:9092"]'
       KAFKA_CLIENT_ID: git-pull-request
       KAFKA_GROUP_ID: amplication-git-pull-request-service
-      GENERATE_PULL_REQUEST_TOPIC: git.internal.pull-request.request.0
+      GENERATE_PULL_REQUEST_TOPIC: git.internal.pull-request.request.1
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/amplication/values-common.yaml:183]: Changed. I pulled the folder of the helm chart from my previous branch and this is a value from it. not relevant changed it to '0'.

File: helm/charts/amplication/values-mor.yaml [hunk 5]
```diff
@@ -42,11 +53,64 @@ amplication-server:
       DOCKER_HOST: tcp://docker:2375
 
       #GitHub App
-      GITHUB_APP_APP_ID: "195638"
-      GITHUB_APP_CLIENT_ID: "Iv1.7c4ec6671b7c8cd7"
-      GITHUB_APP_INSTALLATION_URL: "https://github.com/apps/amplicaiton-server-staging-os/installations/new?state={state}"
+      GITHUB_APP_APP_ID: "215215"
+      GITHUB_APP_CLIENT_ID: "Iv1.a8ac08d4d3788314"
+      GITHUB_APP_INSTALLATION_URL: "https://github.com/apps/mor-app/installations/new?state={state}"
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/amplication/values-mor.yaml:58]: I saw that values-mor is not updated with the rest of the custom-values (like Shimi & Amit) if it doesn't relevant for now I'll restore the values-mor from 'master'.

File: helm/charts/amplication/values-mor.yaml [hunk 6]
```diff
@@ -1,27 +1,38 @@
...
+      REACT_APP_GITHUB_CONTROLLER_LOGIN_URL: "https://server-mor.amplication-dev.com/github"
+      REACT_APP_STORAGE_GATEWAY_URL: https://server-mor.amplication-dev.com
 
 amplication-server:
-  image:
-      repository: 407256539111.dkr.ecr.us-east-1.amazonaws.com/amplication-server
   postgres:
     install: false
   ingress:
     enabled: true
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/amplication/values-mor.yaml:21]: Done

File: helm/charts/amplication/values-prod.yaml [hunk 7]
```diff
@@ -46,7 +41,6 @@ amplication-server:
       DISABLE_EVENT_TRACKING: "1"
 
       #KAFKA
-      GENERATE_PULL_REQUEST_TOPIC: git.internal.pull-request.request.0
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/amplication/values-prod.yaml:49]: restored.

File: helm/charts/amplication/values-staging-os.yaml [hunk 8]
```diff
@@ -108,24 +109,20 @@ amplication-git-pull-service:
...
 
 amplication-git-pull-request-service:
-  pvc:
-    name: staging-os-efs-claim
   config:
     env:
       KAFKA_BROKERS: '["b-2.amplication-useast1-s.0ne4zw.c22.kafka.us-east-1.amazonaws.com:9092","b-1.amplication-useast1-s.0ne4zw.c22.kafka.us-east-1.amazonaws.com:9092"]'
       KAFKA_CLIENT_ID: git-pull-request-service
       KAFKA_GROUP_ID: git-pull-request-service-group
-      GENERATE_PULL_REQUEST_TOPIC: git.internal.pull-request.request.0
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/amplication/values-staging-os.yaml:118]: It's already in values-common and it doesn't change through environments.

File: helm/charts/services/amplication-git-pull-request-service/values.yaml [hunk 9]
```diff
@@ -33,9 +33,4 @@ service:
...
 
 ingress:
-  enabled: false
-
-pvc:
-  storageRequest: 1Gi
-  storageName: efs-sc
-  name: efs-claim
\ No newline at end of file
+  enabled: false
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/services/amplication-git-pull-request-service/values.yaml]: Done

File: helm/charts/services/amplication-git-pull-service/Chart.yaml [hunk 10]
```diff
@@ -6,7 +6,7 @@ dependencies:
 
 - name: postgresql
   alias: postgres
-  version: "10.5.2"
+  version: "10.16.2"
   repository: "https://charts.bitnami.com/bitnami"
   condition: postgres.install
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/services/amplication-git-pull-service/Chart.yaml:11]: No, it's the bug that we can't put nested charts inside another chart as I mentioned above.

File: helm/charts/services/amplication-git-pull-service/values.yaml [hunk 11]
```diff
@@ -34,11 +34,6 @@ service:
 ingress:
   enabled: false
```
[the remark this answers is not in the document]
Comment (shaharblanksela) [reply helm/charts/services/amplication-git-pull-service/values.yaml:35]: Done

**Pull request merged**